### PR TITLE
feat: NFTSDK-400 get transfers by block hash

### PR DESCRIPTION
**Pull request by sahar-fehri**

### CODE REVIEW

File: src/lib/Api/api.ts [hunk 1]
```diff
@@ -136,18 +141,36 @@ export default class Api {
...
-    if (!opts.blockHashNumber) {
+    if (!opts.blockNumber) {
       log.throwMissingArgumentError(Logger.message.invalid_block_number, {
         location: Logger.location.SDK_GETTRANSFERSBYBLOCKNUMBER,
       });
     }
 
     const apiUrl = `${this.apiPath}/nfts/block/transfers`;
-    const { data } = await this.httpClient.get(apiUrl, { blockHashNumber: opts.blockHashNumber });
+    const { data } = await this.httpClient.get(apiUrl, { blockHashNumber: opts.blockNumber });
```
Comment (salimtb) [inline src/lib/Api/api.ts:155]: cursor should be passed as parameter as well

File: src/lib/Api/api.ts [hunk 2]
```diff
@@ -136,18 +141,36 @@ export default class Api {
...
+   */
+  async getTransfersByBlockHash(opts: GetTransfersByBlockHashOptions): Promise<TransfersDTO> {
+    if (!opts.blockHash) {
+      log.throwMissingArgumentError(Logger.message.invalid_block_hash, {
+        location: Logger.location.SDK_GETTRANSFERSBYBLOCKHASH,
+      });
+    }
+
+    const apiUrl = `${this.apiPath}/nfts/block/transfers`;
+    const { data } = await this.httpClient.get(apiUrl, { blockHashNumber: opts.blockHash });
```
Comment (salimtb) [inline src/lib/Api/api.ts:173]: same, cursor is a query param

File: src/lib/Api/api.ts (hunk 1 above)
Comment (sahar-fehri) [reply src/lib/Api/api.ts]: thnx i missed that

File: e2e/sdk.ERC1155.e2e.test.ts [hunk 3]
```diff
@@ -296,7 +296,7 @@ describe('SDK - ERC1155 - contract interaction (deploy, load and mint)', () => {
     );
 
     const transferList3: any = await sdk.api.getTransfersByBlockNumber({
-      blockHashNumber: transfer[0].blockNumber,
+      blockNumber: transfer[0].blockNumber,
     });
     expect(
       transferList3.transfers.filter((tx: any) => tx.transactionHash === txHash.hash)[0],
```
Comment (javiergarciavera) [inline e2e/sdk.ERC1155.e2e.test.ts:302]: you could add here a check for your endpoint as well, which would be coping the lines from 298 to 303 and add them right after replacing getTransfersByBlockNumber with **getTransfersByBlockHash**

**Review by javiergarciavera — APPROVED** [review]

**Review by javiergarciavera — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Pull request merged**